### PR TITLE
[border-agent] name meshcop service after "VendorName_ProductName" 

### PR DESCRIPTION
**Pull request by superwhd**
This PR aims to fix https://github.com/openthread/ot-br-posix/issues/735. 

The recommended format of service instance is `VendorName ProductName._meshcop._udp_.local.`. However, it's quite tricky to handle the space in service name properly. Also, the space in the name will be converted to `\032` after the registration, which is not a very elegant form. Therefore, I chose `VendorName_ProductName._meshcop._udp_.local.` as the default service name.

This PR should be merged together with https://github.com/openthread/openthread/pull/6894.

### CODE REVIEW

**Comment by superwhd** [comment]
The `Depends-On` in the description doesn't seem to let CI patch the depended PR before running workflows. Is there any way to let CI run tests with both PRs patched?

**Comment by jwhui** [comment]
> The `Depends-On` in the description doesn't seem to let CI patch the depended PR before running workflows. Is there any way to let CI run tests with both PRs patched?

@superwhd , the `Depends-On` requires the text to be of the form `openthread/openthread#6865`, rather than the `https://github.com/openthread/openthread/pull/6865` link. I have updated the text above and kicked GitHub Actions tests.

**Comment by superwhd** [comment]
@jwhui Thanks! 

However, it seems the test case stayed unchanged after specifying `Depends-On`. I was expecting [this block](https://github.com/openthread/openthread/pull/6865/files#diff-e67178cb8388ffced2417ed0ad9519b2b4099e9dd20f26a1c8fc814466feaa03R229) would be patched so the test case could be passed.

**Comment by jwhui** [comment]
> However, it seems the test case stayed unchanged after specifying `Depends-On`. I was expecting [this block](https://github.com/openthread/openthread/pull/6865/files#diff-e67178cb8388ffced2417ed0ad9519b2b4099e9dd20f26a1c8fc814466feaa03R229) would be patched so the test case could be passed.

@bukepo , can you help take a look?

**Comment by bukepo** [comment]
we use submodule in this project, and depends-on doesn't work on this project yet. We can get OpenThread merged first and get back to this pr later. Thoughts?

**Comment by superwhd** [comment]
> we use submodule in this project, and depends-on doesn't work on this project yet. We can get OpenThread merged first and get back to this pr later. Thoughts?

SGTM. Will do.

File: src/agent/border_agent.hpp [hunk 1]
```diff
@@ -49,6 +49,18 @@
...
+#ifndef OTBR_VENDOR_NAME
+#define OTBR_VENDOR_NAME "OpenThread"
+#endif
+
+#ifndef OTBR_PRODUCT_NAME
+#define OTBR_PRODUCT_NAME "BorderRouter"
+#endif
+
+#ifndef OTBR_MESHCOP_SERVICE_NAME
+#define OTBR_MESHCOP_SERVICE_NAME OTBR_VENDOR_NAME "_" OTBR_PRODUCT_NAME
```
Comment (wgtdkp) [inline src/agent/border_agent.hpp:61]: There will be name conflicts if two OTBRs are placed on the same network link. mDNS Responder can auto rename the conflicted names, but avahi seems not. See https://github.com/openthread/ot-br-posix/issues/826.

Comment (superwhd) [reply src/agent/border_agent.hpp:61]: I think this issue can be regarded as independent of the current PR.

For the issue #826, I can think of two ways to fix it:
- Generating unique service names (e.g. using random digits for the suffix) before registering the service.
- Leveraging avahi's utils to handle name conflicts. There's an example in their doc: https://www.avahi.org/doxygen/html/client-publish-service_8c-example.html.

**Review by wgtdkp — APPROVED** [review]
LGTM :+1:

File: src/agent/border_agent.cpp [hunk 2]
```diff
@@ -63,7 +63,8 @@
 
 namespace otbr {
 
-static const char kBorderAgentServiceType[] = "_meshcop._udp"; ///< Border agent service type of mDNS
+static const char kBorderAgentServiceType[] = "_meshcop._udp";           ///< Border agent service type of mDNS
+static const char kBorderAgentServiceName[] = OTBR_MESHCOP_SERVICE_NAME; ///< Border agent service name of mDNS
```
Comment (simonlingoogle) [inline src/agent/border_agent.cpp:67]:
```suggestion
static const char kBorderAgentServiceInstanceName[] = OTBR_MESHCOP_SERVICE_INSTANCE_NAME; ///< Border agent service name of mDNS
```
Prefer to add `Instance` for clarity. 
I think `service name` is quite misleading. RFC 6763 refers to service types (e.g. `_meshcop._udp`) as the service names, and always use `instance name` for service instance names.

**Review by simonlingoogle — APPROVED** [review]
LGTM :+1:

**Review by jwhui — APPROVED** [review]

**Pull request merged**